### PR TITLE
Add ASCII table with description for CLI tools

### PR DESCRIPTION
**Pull request by yankeexe**
There are lots of great tools on arkade but it's hard to get context on what they are used for.

Since arkade is a marketplace it is essential to have a description of the products it offers.

Addresses #427

Signed-off-by: yankeexe <yankee.exe@gmail.com>

<details>
<summary><b>Preview of New change: Linux</b></summary>

</details>

<details>
<summary><b>Preview of New change: Windows [Powershell, CMD, Git Bash]</b></summary>

</details>

## Description

* Added a *Description* value to the `Tool` struct to incorporate description for the tool. 
* Added Description for all the tools. 
* Render ASCII table when *no argument* is passed to the `arkade get` command.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by dirien** [comment]
Good feature! Hope it gets merged!

**Comment by yankeexe** [comment]
@alexellis  Can you please review this? :slightly_smiling_face:

**Comment by alexellis** [comment]
Thanks for the interest. Usually, contributors should wait for approval on a design before sending a PR, but in this instance I am fine with reviewing it and giving feedback.

The gif is helpful and looks very cool :+1: 

Have you tested this library with either Git Bash or Windows cmd.exe? It may be that with Windows, you'll find the library may not work as expected.

For the progress-bar work added by @rberrelleza, we spent time testing various OSes.

Looking at [olekukonko's repo](https://github.com/olekukonko/tablewriter), most parts of it haven't been updated for 3-7 years (especially the functionality itself).

From the open issues and questions, most have not received an answer from the maintainer, [like this one from Dec 2020](https://github.com/olekukonko/tablewriter/issues/177) on a fix for the go.mod file.

Were there other table writers that you considered, or that you've seen elsewhere? How do they compare?

File: pkg/get/tools.go [hunk 1]
```diff
@@ -24,9 +24,10 @@ func MakeTools() Tools {
 
 	tools = append(tools,
 		Tool{
-			Owner: "openfaas",
-			Repo:  "faas-cli",
-			Name:  "faas-cli",
+			Owner:       "openfaas",
+			Repo:        "faas-cli",
+			Name:        "faas-cli",
+			Description: "Official CLI for OpenFaaS.",
```
Comment (alexellis) [inline pkg/get/tools.go:30]: Thanks for adding all these Descriptions, that must have taken you a long time. It's appreciated.

**Review by alexellis — COMMENTED** [review]
Sending a review since you pinged me on Slack today for comment.

File: pkg/get/tools.go (hunk 1 above)
Comment (yankeexe) [reply pkg/get/tools.go:30]: Welcome, it was pretty fun to explore all the tools offered by arkade.

**Comment by yankeexe** [comment]
> Were there other table writers that you considered or that you've seen elsewhere? How do they compare?

@alexellis, I haven't faced any of the issues from the repo. 

I did compare few Terminal UI tools and ASCII table rendering packages like [pterm table](https://github.com/pterm/pterm#table), [go-pretty](https://github.com/jedib0t/go-pretty/tree/main/table), and [simple table](https://github.com/alexeyco/simpletable).

I wanted a minimal package with multiple table formatting and color options, which I found on [tablewriter](https://github.com/olekukonko/tablewriter). Also, auto wrapping text on rows is pretty handy. However, with other pkgs, wrapping text is tricky, lacks table formatting. 

Also, `cmd` on Windows, doesn't render the colors properly so I disabled the colors if it's running on windows, please check the description for updated screenshots. 🙂

**Comment by rberrelleza** [comment]
@yankeexe looks cool! One issue I ran into with the progress bar was when the user runs on WSL, since it's linux, but the terminal is a windows one. Might want to check on that.

On github.com/okteto/okteto we use github.com/fatih/color to output colorized strigs, and that works well on Windows, in case you want to check it out.

**Comment by yankeexe** [comment]
> @yankeexe looks cool! One issue I ran into with the progress bar was when the user runs on WSL, since it's linux, but the terminal is a windows one. Might want to check on that.
> 
> On https://github.com/okteto/okteto we use https://github.com/fatih/color to output colorized strigs, and that works well on Windows, in case you want to check it out.

thank you @rberrelleza I checked it on Linux terminal as well (Manjaro Konsole) app. I will give the color package a try and test it on Windows. Thank you for the suggestion 🙂

**Comment by alexellis** [comment]
> Looking at olekukonko's repo, most parts of it haven't been updated for 3-7 years (especially the functionality itself).
> 
> From the open issues and questions, most have not received an answer from the maintainer, like this one from Dec 2020 on a fix for the go.mod file.

This is still a concern for me and I don't feel like you've answered the comments, however if I merge this and then the repository gets archived or has a major security issue, would you be willing to find a replacement?

**Comment by alexellis** [comment]
Let's go ahead and get this merged then, we just need a rebase because I merged something by @dirien this morning that was also in the queue with your PR.

Alex

**Comment by alexellis** [comment]
I'm now seeing: `This branch cannot be rebased due to conflicts`

If you squash your two commits into one, then run:

```
git pull origin master --rebase
```

It will prompt you on each conflict and then put your changes over the top.

Between fixing each merge conflict you need to run `git add . && git rebase --continue`

**Comment by alexellis** [comment]
Thank you 🎉

**Pull request merged**